envi/memcanvas/__init__.py

```diff
@@ -383,6 +383,40 @@ def __str__(self):
         return self.strval
 
 
+class FileBackedMemoryCanvas(StringMemoryCanvas):
+    '''
+    FileBackedMemoryCanvas writes output to a file of your choosing.  It writes
+    out the data to the file after `rollwindow` number of calls to `addText()`.
+
+    NOTE: when done writing to the canvas, be certain to call `flush()` or you
+            will lose data.
+    '''
+    def __init__(self, filename, mem, syms=None, append=True, rollwindow=100):
+        StringMemoryCanvas.__init__(self, mem, syms)
+        if append:
+            self.fd = open(filename, 'ab')
```

Comment:
so... ummm.... no we don't...
only in symcache and envi.config

all the other calls to open(), in all the parsers, vivserver, vdb, mpfile, basicfile, etc...  we don't.  not even the code this is replacing (which writes to a file).

but we can if you like?  is this a standard you'd like to make?  i'm not opposed, i just want to make sure we're dealing with the same facts, and that you're not thinking of synapse or some other codebase.

Reply:
Yea. That's what I'm referring to. We don't do it for the binary parsers because it makes no sense there because utf-8 makes no sense there. But for the straight text ones we do, or at least should. We might have missed some spots, but we can mop those up.

Reply:
that's fair.
i can change that.  the original code used bytes ('wb') but we can make this a utf-8-encoded file.  however, if by chance the output uses any high bits, it's not as good.

Reply:
So do you want any of the canvas text to include any output, including any unadorned raw bytes, or should it just be strings?

Because before, any strings were encoded to utf-8 before being written, so we're more just enforcing in `open` what we already had before and keeping things consistent. Right now the FileBacked canvas doesn't do any encoding, which is a change from before. Which I'm not against, but we should be explicit about that since it is a subtle but important API change.

Reply:
this actually isn't a change in behavior.  compare envi.cli's `do_saveout` and look here at FileBacked's `flush()`.  they both write utf-8 encoded strings.

we can set the file encoding as well.

Reply:
done.  had to change both the open() and the write()